todo_app/test_e2e/test_app.py

```diff
@@ -0,0 +1,91 @@
+import os
+import pytest
+from threading import Thread
+from time import sleep
+from selenium import webdriver
+from selenium.webdriver import FirefoxOptions
+from dotenv import load_dotenv
+from todo_app import app
+import requests
+
+headers = {
+  "Accept": "application/json"
+}
+
+query = {
+  'key': os.getenv('TRELLO_API_KEY'),
+  'token': os.getenv('TRELLO_API_TOKEN')
+}
```

Comment:
I think this isn't quite working - you're reading out these environment variables before your test has had a chance to actually load them into memory! As far as I can tell the structure for this test is right but it's not quite working - the easy fix in this file would be to duplicate this query into each function it's used, or swap this to a function, e.g.:
```
def get_query():
    return {
      'key': os.getenv('TRELLO_API_KEY'),
      'token': os.getenv('TRELLO_API_TOKEN')
    }
```
And then call that rather than trying to access this variable

You'd need to do a similar thing in the `trello_items.py` file as well, as I suspect that's suffering from the same situation

Feel free to reach out if it's not clear what I mean here, as it's a fairly subtle issue!

Reply:
yes, you are right I didn't have time to get it fully working and I was finding problems working on WSL... but maybe It was a red herring. Any way, I'll try to get it working in the future.